### PR TITLE
Backport e2e-test of the latest tendermint main branch

### PR DESCRIPTION
**Pull request by tnasu**
## Description

`e2e-test` is old and it makes too many failures each PR and Push. I investigated the error logs of `e2e-test`, and it shows **light01** failure due to starting delay. **light01** depends on the witness("**validator01**", "**validator02**", "**validator03**") and **validator02** depends on the seeds("**seed02**"). And the log shows that "**seed02**" is in the process of starting normally. Therefore, "**validator02**" and "**light01**" were also starting up and did not reach the expected Height within the expected time, resulting in an error. So "**seed02**" is removed.

* light01 happened an error
```
light01        | I[2022-12-21|02:55:43.981] Downloading trusted light block using options module=main 
light01        | I[2022-[12](https://github.com/line/ostracon/actions/runs/3738979308/jobs/6360920253#step:10:13)-21|02:55:43.985] error comparing first header with witness. You may want to consider removing the witness module=main err="light block not found"
light01        | panic: runtime error: invalid memory address or nil pointer dereference
light01        | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc220b4]
light01        | 
light01        | goroutine 30 [running]:
light01        | github.com/line/ostracon/light/provider/http.(*http).LightBlock(0xc000ba1a40, {0x153f440, 0xc000b64c40}, 0x0)
light01        | 	/src/ostracon/light/provider/http/http.go:86 +0x[13](https://github.com/line/ostracon/actions/runs/3738979308/jobs/6360920253#step:10:14)4
light01        | github.com/line/ostracon/light.(*Client).getTargetBlockOrLatest(0xc000ba1a40?, {0x153f440, 0xc000b64c40}, 0x3e8, {0x153c3a0, 0xc000ba1a40})
light01        | 	/src/ostracon/light/detector.go:382 +0x3c
light01        | github.com/line/ostracon/light.(*Client).compareNewHeaderWithWitness(0xc00003e840, {0x153f440, 0xc000b64c40}, 0xc00005ff78?, 0xc000537a70, {0x153c3a0, 0xc000ba1a40}, 0x0)
light01        | 	/src/ostracon/light/detector.go:136 +0x285
light01        | created by github.com/line/ostracon/light.(*Client).compareFirstHeaderWithWitnesses
light01        | 	/src/ostracon/light/client.go:1[14](https://github.com/line/ostracon/actions/runs/3738979308/jobs/6360920253#step:10:15)5 +0x96b
```
* validator02 is starting now, it's not started, so light01 cannot get a block
```
validator02    | I[2022-12-21|02:55:26.107] received complete proposal block             module=consensus height=1000 hash=158E96A848B4AF973BAFAF7453B3077DEDE33044D49C327B44A853A101C86E88
validator02    | I[2022-12-21|02:55:29.105] Timed out                                    module=consensus dur=3s height=1000 round=0 step=RoundStepPropose
validator02    | I[2022-12-21|02:55:50.053] Starting Peer service                        module=p2p peer=2551a13ed720101b271a5df4816d1e4b3d3bd133@[fd80:b10c::6]:52926 impl="Peer{MConn{[fd80:b10c::6]:52926} 2551a13ed720101b271a5df4816d1e4b3d3bd133 in}"
validator02    | I[2022-12-21|02:55:50.053] Starting MConnection service                 module=p2p 
```
* seed02 is started at "02:55:50", so validator02 cannot start
```
seed02         | I[2022-12-21|02:55:50.602] commit is for a block we do not know about; set ProposalBlock=nil module=consensus height=1000 commit_round=1 proposal= commit=B0278D2A75529747D37EFB72D40A9A30AB6EEC716BAE3C490F3DB45F9246428B
seed02         | I[2022-12-21|02:55:50.703] received complete proposal block             module=consensus height=1000 hash=B0278D2A75529747D37EFB72D40A9A30AB6EEC716BAE3C490F3DB45F9246428B
seed02         | I[2022-12-21|02:55:50.704] finalizing commit of block                   module=consensus height=1000 hash=B0278D2A75529747D37EFB72D40A9A30AB6EEC716BAE3C490F3DB45F9246428B 
```

Based on: https://github.com/tendermint/tendermint/commit/b1fbe26ffadae6e4010a71bd75767c6680d1bf71
* e2e-test: https://github.com/tendermint/tendermint/tree/b1fbe26ffadae6e4010a71bd75767c6680d1bf71/test/e2e
* e2e-test history: https://github.com/tendermint/tendermint/commits/b1fbe26ffadae6e4010a71bd75767c6680d1bf71/test/e2e

Refs:
* https://github.com/tendermint/tendermint/pull/9292/files#r950183182

Remarks: (Excludes some of the new features)
* https://github.com/tendermint/tendermint/pull/8695
* https://github.com/tendermint/tendermint/pull/9122
* https://github.com/tendermint/tendermint/pull/9146
* https://github.com/tendermint/tendermint/pull/9259
* https://github.com/tendermint/tendermint/pull/9287
* https://github.com/tendermint/tendermint/pull/9335
* https://github.com/tendermint/tendermint/pull/9655
* https://github.com/tendermint/tendermint/pull/9830

### CODE REVIEW

File: libs/log/ocfmt_logger_test.go [hunk 1]
```diff
@@ -61,17 +61,17 @@ func TestOCFmtLogger(t *testing.T) {
 	assert.Regexp(t, regexp.MustCompile(`N\[.+\] unknown \s+ hash=74657374206D65\n$`), buf.String())
 }
 
-func BenchmarkOCFmtLoggerSimple(b *testing.B) {
-	benchmarkRunnerKitlog(b, log.NewOCFmtLogger(ioutil.Discard), baseMessage)
+func BenchmarkTMFmtLoggerSimple(b *testing.B) {
```
Comment (tnasu) [inline libs/log/ocfmt_logger_test.go:64]:
```suggestion
func BenchmarkOCFmtLoggerSimple(b *testing.B) {
```

File: libs/log/ocfmt_logger_test.go [hunk 2]
```diff
@@ -61,17 +61,17 @@ func TestOCFmtLogger(t *testing.T) {
...
+	benchmarkRunnerKitlog(b, log.NewOCFmtLogger(io.Discard), baseMessage)
 }
 
-func BenchmarkOCFmtLoggerContextual(b *testing.B) {
-	benchmarkRunnerKitlog(b, log.NewOCFmtLogger(ioutil.Discard), withMessage)
+func BenchmarkTMFmtLoggerContextual(b *testing.B) {
```
Comment (tnasu) [inline libs/log/ocfmt_logger_test.go:68]:
```suggestion
func BenchmarkOCFmtLoggerContextual(b *testing.B) {
```

File: test/e2e/generator/main_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import (
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestNewCLI(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "runner")
```
Comment (tnasu) [inline test/e2e/generator/main_test.go:11]:
```suggestion
	tempDir, err := ioutil.TempDir("", "generator")
```

**Comment by kokeshiM0chi** [comment]
Simple question. Please tell me what you think.
Q1. About fixing timing errors
Since **validator02** depends on **seed02**, **light01**, which depends on **validator02**, cannot get a block if **seed02** is delayed. So, **validator02** is set to depend on **seed01** like **validator03** and **validator01**.

Is it a change that lowered the probability of failure due to delay with the above change?

Q2. About post-commit errors
https://github.com/line/ostracon/actions/runs/3749344661/jobs/6367896870#step:9:12

>E[2022-12-21|12:55:35.620] timed out waiting for seed01 to reach height 0

Looking at this error, does this mean that if seed01 is late it can error out as well?

**Comment by tnasu** [comment]
@kokeshiM0chi 

> Q1

Yes, it does. Because we cannot control the network condition on GitHub Actions.

> Q2

Yes, it does. But, if `seed01` happens an error, it returns an error immediately and we can re-run e2e-test immediately. `light02`'s error is so late and waits a long time.

**Review by ulbqb — APPROVED** [review]
LGTM

**Review by kokeshiM0chi — APPROVED** [review]
LGTM

**Comment by torao** [comment]
Tip: The `test/maverick` using the Byzantine node has been removed in this PR. In subsequent modifications, it's not necessary to also modify the maverick package when modifying `consensus` or `node` modules.

**Pull request merged**